### ISSUE TITLE
BUG: argc sometimes larger than the number of arg fields in EXECVE records

### ISSUE THREAD

**Issue by stevegrubb** [linux-audit/audit-kernel#99]
This is a continuation from https://github.com/linux-audit/audit-userspace/issues/57
It was reported that argc does not match argument count in the EXECVE record. For example:
type=EXECVE msg=audit(1469116325.662:1707): argc=2 a0="/bin/echo"

This can be looked for in existing logs with:
```
#!/usr/bin/env python2

import glob

files = glob.glob("/var/log/audit/*")

for fname in files:
    with open(fname, "r") as f:
            for line in f:
                if 'type=EXECVE' in line:
                    try:
                        # extract sub-string after argc=
                        substring = line.split("argc=",2)[1]
                        args=substring.partition(' ')[0]
                        actual_args = substring.count(' ')
                        if (int(args) != actual_args):
                            print line 
                    except:
                        print "Ran into an error: " + line
```

Problem seems to be related to empty arguments. This can be reproduced as follows:

autrace /bin/ls "" "/etc"
ausearch --start recent -m execve -i | grep EXECVE
type=EXECVE msg=audit(10/03/2018 13:04:03.208:1391) : argc=3 a0=/bin/ls a2=/etc

**Comment by rgbriggs** [comment]
Paul, please assign to me.

**Comment by pcmoore** [comment]
@rgbriggs Done.  I believe I touched this code last, so I'm happy to look into this if you don't have time.

**Comment by rgbriggs** [comment]
On 2018-10-03 14:04, Paul Moore wrote:
> @rgbriggs Done.  I believe I touched this code last, so I'm happy to look into this if you don't have time.

Yes, I believe you did, but I've already looked at the code and have an
idea what to do.  I'm blocked on upstream review, so I'd rather you
spent your time there.

**Comment by pcmoore** [comment]
Merged into audit/next via ea956d8be91edc702a98b7fe1f9463e7ca8c42ab
